neqo-crypto/src/aead.rs

```diff
@@ -158,6 +191,41 @@ impl RealAead {
         }?;
         Ok(&output[0..(l.try_into()?)])
     }
+
+    /// Decrypt a ciphertext in place.
+    ///
+    /// Note that NSS insists upon having extra space available for decryption, so
+    /// the buffer for `output` should be the same length as `input`, even though
+    /// the final result will be shorter.
```

Comment:
I would move this note to a comment on the last argument to the function call below.  This API doesn't need to bother API users with this particular detail.